src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -67,12 +78,48 @@ class ActivityFeed extends React.Component<Props, State> {
         if (hasLoaded || hasMoreItems || hasNewItems || hasInputOpened) {
             this.resetFeedScroll();
         }
+
+        if (currCurrentUser !== prevCurrentUser || currFeedItems !== prevFeedItems) {
+            this.checkAndLogRenderMetrics(currCurrentUser, currFeedItems);
+        }
+    }
+
+    /**
+     * Log render and load state timing events.
+     *
+     * @param {*} currentUser
+     * @param {*} feedItems
+     */
+    checkAndLogRenderMetrics(currentUser?: User, feedItems?: FeedItems) {
+        const { logger } = this.props;
+        const hasLoggedRender = logger.hasLoggedEvent(ACTIVITY_FEED_COMPONENT, EVENT_TIME_TO_RENDER);
+        if (!hasLoggedRender) {
+            if (currentUser && feedItems) {
+                window.performance.mark(ACTIVITY_FEED_TAGS.TimeToRender);
+                logger.logTimeMetric(
+                    ACTIVITY_FEED_COMPONENT,
+                    EVENT_TIME_TO_RENDER,
+                    ACTIVITY_SIDEBAR_TAGS.Initialized,
+                    ACTIVITY_FEED_TAGS.TimeToRender,
+                    true,
+                );
+            } else if (!logger.hasLoggedEvent(ACTIVITY_FEED_COMPONENT, EVENT_TIME_TO_LOAD_STATE)) {
+                window.performance.mark(ACTIVITY_FEED_TAGS.TimeToLoadState);
```

Comment:
I am unsure if this is accurate. I think you are trying to measure time to loading indicator displayed, but it will be displayed on first render which will occur before this. Should that be from from constructor to componentDidMount?

Reply:
I'm going to be honest, if loading is occurring in first render pass, I'm not sure we're going to get any use out of this metric. I'm going to remove it, to simplify for now. We can determine where loading state is useful to track, later.